src/telliot_core/contract/contract.py

```diff
@@ -156,17 +189,33 @@ async def write(
     async def write_with_retry(
         self,
         func_name: str,
-        gas_price: int,
         extra_gas_price: int,
         retries: int,
         gas_limit: int,
+        legacy_gas_price: Optional[int] = None,
+        max_priority_fee_per_gas: Optional[int] = None,
+        max_fee_per_gas: Optional[int] = None,
         **kwargs: Any,
     ) -> Tuple[Optional[AttributeDict[Any, Any]], ResponseStatus]:
         """For submitting any contract transaction. Retries supported!
 
         gas_price measured in gwei
 
         """
+        if (legacy_gas_price is not None) and (
```

Comment:
You probably don't need to re-check/duplicate this code.  It would get rejected by the contract.write call.